Source/GTMSessionFetcher.h

```diff
@@ -608,6 +610,51 @@ NSData *_Nullable GTMDataFromInputStream(NSInputStream *inputStream, NSError **o
 }  // extern "C"
 #endif
 
+// Completion handler passed to -[GTMFetcherDecoratorProtocol fetcherWillStart:completionHandler:].
+typedef void (^GTMFetcherDecoratorFetcherWillStartCompletionHandler)(NSURLRequest *_Nullable,
+                                                                     NSError *_Nullable);
+
+// Allows intercepting a request and optionally modifying it throughout the lifetime of the
+// request. See `-[GTMSessionFetcherService addDecorator:]` and `-[GTMSessionFetcherService
+// removeDecorator:]`.
+@protocol GTMFetcherDecoratorProtocol <NSObject>
+
+// Invoked just before a fetcher's request starts.
+//
+// After the decorator's work is complete, the decorator must invoke `handler(request, error)`
+// either synchronously or asynchronously (on any queue).
+//
+// If `error` is non-nil, then the fetcher is stopped with the given error, and any further
+// decorators' `-fetcherWillStart:completionHandler:` methods are not invoked.
+//
+// Otherwise, the decorator may use `[fetcher.request mutableCopy]`, make changes to the mutable
+// copy of the request, and pass the result to the handler via the `request` parameter.
```

Comment:
We should document what happens if the `request` parameter to the callback is `nil`. Perhaps something along the lines of "If no changes to the request are being made, pass `nil` to the callback for the `request` parameter." (happy to have that word-smithed)?

Reply:
Thanks, done.